hw/ip/spi_host/doc/_index.md

```diff
@@ -0,0 +1,762 @@
+If the last few received bytes do not fill an entire RXDATA word, they will be posted to RXDATA at the end of the command and any unneeded bytes the end of the RXDATA word will be left as zeros.
+
+The following waveform illustrates an example SPI transaction, where neither the data transmitted nor the data received fit into an even number of 32-bit words.
+In this example, `A[31:0]` and `B[31:0]`, have been previously loaded into {{< regref "TXDATA" >}}, and afterwards one word, `X[31:0]`, is available in {{< regref "RXDATA" >}}.
+All data in the waveform is transferred using 32-bit instructions.
+
+{{< wavejson >}}
+{signal: [
+  ["ByteOrder=0",
+  {name: "sd[0]", wave: "x222222223344552233z.22x", data: ["A[31]", "A[30]", "A[29]", "A[28]", "A[27]","A[26]", "A[25]", "A[24]", "A[23]", "A[19]", "A[15]", "A[11]",
+                                                         "A[7]", "A[3]", "B[31]", "B[27]", "B[23]", "B[19]", "X[31]", "X[27]"]},
+  {name: "sd[1]", wave: "xz.......3344552233z.22x", data: ["A[22]", "A[18]", "A[14]", "A[10]", "A[6]", "A[2]", "B[30]", "B[26]", "B[22]", "B[18]", "X[30]", "X[26]"]},
+  {name: "sd[2]", wave: "xz.......3344552233z.22x", data: ["A[21]", "A[17]", "A[13]", "A[9]", "A[5]", "A[1]", "B[29]", "B[25]", "B[21]", "B[17]", "X[29]", "X[25]"]},
+  {name: "sd[3]", wave: "xz.......3344552233z.22x", data: ["A[20]", "A[16]", "A[12]", "A[8]", "A[4]", "A[0]", "B[28]", "B[24]", "B[20]", "B[16]", "X[28]", "X[24]"]},
+],
+   {name:""},
+   ["ByteOrder=1",
+  {name: "sd[0]", wave: "x555555554433225544z.55x", data: ["A[7]", "A[6]", "A[5]", "A[4]", "A[3]", "A[2]", "A[1]", "A[0]",
+                                                           "A[15]", "A[11]", "A[23]", "A[19]", "A[31]", "A[27]", "B[7]", "B[3]", "B[15]", "B[11]", "X[7]", "X[3]"]},
+  {name: "sd[1]", wave: "xz.......4433225544z.55x", data: ["A[14]", "A[10]", "A[22]", "A[18]", "A[30]", "A[26]", "B[6]", "B[2]", "B[14]", "B[10]", "X[6]","X[2]"]},
+  {name: "sd[2]", wave: "xz.......4433225544z.55x", data: ["A[13]", "A[9]", "A[21]", "A[17]", "A[29]", "A[25]", "B[5]", "B[1]", "B[13]", "B[9]", "X[5]", "X[1]"]},
+  {name: "sd[3]", wave: "xz.......4433225544z.55x", data: ["A[12]", "A[8]", "A[20]", "A[16]", "A[28]", "A[24]", "B[4]", "B[0]", "B[12]", "B[8]", "X[4]", "X[0]"]},
+  ],
+  ],
+  head: {
+    text: "Serial bit ordering for 6 bytes transmitted from FIFO words 'A[31:0]' and 'B[31:0]', and 1 byte received into word 'X[31:0]'",
+  },
+  foot: {
+    text: "Note: SPI_HOST command sent with COMMAND_0.TX1_CNT_0 = 1, .TXN_CNT_0 = 5, .DUMMY_CYCLES_0 = 2, and  .RX_CNT_0=1"
+  }
+}
+{{< /wavejson >}}
+
+
+There is also no restriction on the alignment of transmitted data. 
+The {{< regref "TXDATA" >}} register is actually implemented as a memory window, so that it can support byte-enable signals.
+This means that when copying bytes into {{< regref "TXDATA" >}} from unaligned firmware memory addresses, it is possible to use byte or half-word instructions.
```

Comment:
@tjaychen: this whole section from #5077 has been moved here to help close it.

[Your comment on this line](https://github.com/lowRISC/opentitan/pull/5077#discussion_r571315714) has been carried over here also:

> hmmm.... this could end up becoming a problem.
> We can talk to the software guys a bit more..but i'm pretty sure they really don't like the idea of having to pack in software to 4-byte alignments. I'm pretty certain the ideal would be to treat this write almost like a memory.
> 
> I guess the point you're trying to make is that they don't need to pack, the bytes un-written will simply not be used?
> This could end up wasting a bit of space.. but i guess it's not a huge problem. Let me know if i understood correctly.

Reply:
I think looking at the original thread, this can be closed so long as the FIFO size is increased? Am I right?

Reply:
I'm not quite sure what the text is telling me here:

Are the FIFOs byte-addressable or word-addressable?  L400 above seems to indicate byte-addressability, but L401 below suggests that a byte write may affect an entire word.